nixos/modules/services/monitoring/thanos.nix

```diff
@@ -187,6 +187,10 @@ let
         Make sure this address is routable from other components.
       '';
 
+      grpc-compression = mkParam types.str ''
+        Compression algorithm to use for gRPC requests to other clients.
+      '';
+
```

Comment:
Hello, sorry for the late review.

https://github.com/search?q=repo%3Athanos-io%2Fthanos%20grpc-compression&type=code

* `grpc-compression` option is only available with `query` subcommand
* `receive` subcommand uses `receive.grpc-compression` option

I think you should move the `grpc-compression` parameter in the `query` parameters (~line 337).

What do you think?